subgraph/schema.graphql

```diff
@@ -1,53 +1,30 @@
-type UnlockEntity @entity {
+type Lock @entity {
+  id: ID!
+  address: Bytes!
+  LockManagers: [LockManager!]
+  keys: [Key!] @derivedFrom(field: "lock")
+  createdAtBlock: BigInt
+  version: BigInt
+}
+
+type LockManager @entity {
   id: ID!
-  count: BigInt!
-  lockOwner: Bytes! # address
-  newLockAddress: Bytes! # address
+  address: Bytes!
 }
 
 type Key @entity {
   id: ID!
   lock: Lock!
   keyId: BigInt!
-  owner: KeyHolder!
+  owner: KeyOwner!
+  manager: Bytes
```

Comment:
It should be an address?

Reply:
There are no `Address` type in the schema as it supports only Ethereum types (i.e. Bytes / BigInt / String). The subgraph lib takes care of the casting btw types / address at the AssemblyScript application level https://thegraph.com/docs/en/developing/assemblyscript-api/#support-for-ethereum-types